### DISCUSSION TITLE
Does Aptos supports publish module in module?

### DISCUSSION THREAD

**Discussion by Metal-joker** [aptos-labs/aptos-developer-discussions#148] [Questions] [answered]
### Describe your question in detail. For example how do you get all transactions for a user with the TypeScript SDK?

as title

### What have you tried or looked at?  Or how can we reproduce the error?

Not an error, just a question.

### Which operating system are you using?

N/A

### Which SDK or tool are you using? (if any)

N/A

**Comment by gregnazario** [comment]
Publish module in module?

I'm not sure what that would entail.  Are you trying to call publish module within a module, or are you trying to publish a module inside another module

**Reply by Metal-joker** [reply]
I'm trying to publish a module inside a module, like in solidity, `new SomeContract();`

To add more context:
here's a link to solidity [doc](https://docs.soliditylang.org/en/v0.8.25/contracts.html#creating-contracts)
and heres a quote:

> Contracts can be created “from outside” via Ethereum transactions or from within Solidity contracts.

I want this part `or from within Solidity contracts`.

I check the Aptos docs and find a [handle](https://aptos.dev/reference/move?branch=mainnet&page=aptos-framework/doc/code.md#0x1_code_publish_package_txn) may or may not be solution, I'm not sure.
Also not sure how to get those parameters.
Any insight would be appreciated.

**Reply by gregnazario** [reply]
It would technically be possible, if you use the publish pkg function since it is public.

But, it would be much harder to get a properly compiling package to publish if it isn't just replacing specific bytecode.

Generally the compiler makes this much easier

**Reply by Metal-joker** [reply]
Besides that, is there anyway to construct a type argument inside move?
Let's say I did published a module inside a module, and I want to use that module as type arguments like this:
```
// dummy struct for demo
struct Dummy<T> {};

code::publish_package_txn(x, y, z)
// now I have this modules address and name, 

// is there something iI can do to make this happen?
let dummy = Dummy<module_address::module_name> {};

```

**Reply by gregnazario** [reply, marked as answer]
At the time of publish, no.  It would have to be in the next transaction.
